src/utils/delay.js

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+/**
+ * Invokes func after wait milliseconds
+ *
+ * @param {Function} func The function to delay.
+ * @param {Number} wait The number of milliseconds to delay invocation
+ * @returns {Number} The timer ID
+ */
+export default function delay(func, wait = 0, win = window) {
+  return win.setTimeout(func, Number(wait) || 0);
+}
+
+/**
+ * Cancels an in progress delayed func
+ *
+ * @param {Number} The timer ID
+ */
+export function cancelDelay(id = -1, win = window) {
+  win.clearTimeout(id);
+}
```

Comment:
It seems like the validation/conversion/fallback logic is pretty much how it behaves natively. In other words, if you pass `null` or `NaN` as the timeout to `setTimeout`,  internally it converts it to 0 automatically. `clearTimeout` is similarly lenient. I think we could remove some of the logic:

```
export default function delay(func, wait, win = window) {
  return win.setTimeout(func, wait);
}
```

and `cancelDelay` to:

```
export function cancelDelay(id, win = window) {
  win.clearTimeout(id);
}
```

But at that point I don't know what value the utility provides. I can see how it might be useful when mocking `window.setTimeout()` and `window.clearTimeout()`, but I think `jasmine.clock()` would suffice for those cases?

Reply:
@Aaronius I gave it a second thought and I removed this utility. `setTimeout` is good enough there is no need to abstract it away.